### PR TITLE
[node] http: Update `ClientRequestArgs` and `ServerOptions`

### PR DESCRIPTION
**Pull request by Semigradsky**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/docs/latest-v18.x/api/http.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

Closes #63194

### CODE REVIEW

File: types/node/ts4.8/http.d.ts [hunk 1]
```diff
@@ -113,56 +114,87 @@ declare module 'http' {
...
+         */
         ServerResponse?: Response | undefined;
         /**
-         * Optionally overrides the value of
-         * `--max-http-header-size` for requests received by this server, i.e.
-         * the maximum length of request headers in bytes.
-         * @default 8192
+         * Sets the timeout value in milliseconds for receiving the entire request from the client.
+         * @see Server.requestTimeout for more information.
+         * @default 5000
```
Comment (peterblazejewicz) [inline types/node/ts4.8/http.d.ts:163]:
```suggestion
         * @default 300000
```
Not sure here though:
https://nodejs.org/docs/latest-v18.x/api/http.html#httpcreateserveroptions-requestlistener

File: types/node/http.d.ts [hunk 2]
```diff
@@ -113,56 +114,87 @@ declare module 'http' {
...
+         */
         ServerResponse?: Response | undefined;
         /**
-         * Optionally overrides the value of
-         * `--max-http-header-size` for requests received by this server, i.e.
-         * the maximum length of request headers in bytes.
-         * @default 8192
+         * Sets the timeout value in milliseconds for receiving the entire request from the client.
+         * @see Server.requestTimeout for more information.
+         * @default 5000
```
Comment (peterblazejewicz) [inline types/node/http.d.ts:163]:
```suggestion
         * @default 300000
```
as in other

**Review by peterblazejewicz — APPROVED** [review]
@Semigradsky thanks!
LGTM!

**Comment by Semigradsky** [comment]
> Error in node
Error: Cannot find module '/home/runner/.dts/typescript-installs/4.9/node_modules/typescript'

**Comment by Semigradsky** [comment]
@peterblazejewicz Sorry for disturbing. Maybe you know why tests failed?

**Comment by peterblazejewicz** [comment]
> @peterblazejewicz Sorry for disturbing. Maybe you know why tests failed?

Please review PRs I've flagged you, than let's rebase from master so CI rerun

**Comment by Semigradsky** [comment]
> > @peterblazejewicz Sorry for disturbing. Maybe you know why tests failed?
> 
> Please review PRs I've flagged you, then let's rebase from master so CI rerun

Looks like changes in `@types/conventional-commits-parser` `/conventional-commits-parser-tests.ts` also needed?
```
Error in conventional-commits-parser
Error: /home/runner/work/DefinitelyTyped/DefinitelyTyped/types/conventional-commits-parser/conventional-commits-parser-tests.ts:26:9
ERROR: 26:9  expect  TypeScript@5.0 expected type to be:
  Commit<string | number | symbol>
got:
  Commit
```

**Comment by peterblazejewicz** [comment]
> Looks like changes in `@types/conventional-commits-parser` `/conventional-commits-parser-tests.ts` also needed?

this looks like a flaky test under TS5.0. I got not errors locally, same TS version (@next), rerun CI, faled on this module. Let me look

**Comment by peterblazejewicz** [comment]
I'll ask TS team on DS, I cannot get to problem behind failures, same @next version works on my local setup

**Comment by peterblazejewicz** [comment]
@Semigradsky figured this out: #63201

**Comment by Semigradsky** [comment]
Great!

**Review by peterblazejewicz — APPROVED** [review]

**Comment by Semigradsky** [comment]
Ready to merge

**Pull request merged**